### PR TITLE
Custom reports: make views global

### PR DESCRIPTION
**Pull request by carkom**
this is to convert the view options (legend/summary/labels) to global preferences that apply to all graphs.

### CODE REVIEW

File: packages/desktop-client/src/components/reports/ReportLegend.tsx [hunk 1]
```diff
@@ -31,37 +31,38 @@ function ReportLegend({ legend, groupBy }: ReportLegendProps) {
...
-            <View
-              key={item.name}
-              style={{
-                padding: 10,
-                flexDirection: 'row',
-                alignItems: 'center',
-              }}
-            >
+        {legend &&
+          legend.map(item => {
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/reports/ReportLegend.tsx:35]: **💬 suggestion:** ‏is `legend` optional? you'll want to update the prop type too then to reflect this better.

File: packages/desktop-client/src/components/reports/ReportSummary.tsx [hunk 2]
```diff
@@ -130,7 +130,7 @@ function ReportSummary({
           }}
         >
           <PrivacyFilter blurIntensity={7}>
-            {integerToCurrency(Math.round(average))}
+            {average && integerToCurrency(Math.round(average))}
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/reports/ReportSummary.tsx:133]: **🔨 warning:** ‏if `average <= 0` - it won't show it. Is that expected? Or did you just try to catch `NaN` value (you can use `isNaN` for that)?

File: packages/desktop-client/src/components/reports/ReportLegend.tsx (hunk 1 above)
Comment (carkom) [reply packages/desktop-client/src/components/reports/ReportLegend.tsx]: I've added an error check instead. LMK if this works.

Comment (MatissJanis) [reply packages/desktop-client/src/components/reports/ReportLegend.tsx]: I think we have a misunderstanding. Your previous logic was totally fine, but the component prop type needs to reflect the new reality. Let me explain a bit more in depth.

```
// This component always requires a `test` prop to be set
function MyComponent(props: { test: boolean }) {}


// This component makes `test` prop optional
function MyComponent(props: { test?: boolean }) {}

// This is what you currently have:
type ReportLegendProps = {
  // notice that the `legend` prop is ALWAYS required
  legend: Array<{ name: string; color: string }>;
  groupBy: string;
}

// Instead it should now be
type ReportLegendProps = {
  legend?: Array<{ name: string; color: string }>;
  groupBy: string;
```

**Review by MatissJanis — APPROVED** [review]

**Pull request merged**